### PR TITLE
[Fix] cannot to save the best checkpoint when the key_score is None

### PR DESCRIPTION
**Pull request by LuooChen**
## Motivation

Related Issue: [open-mmlab/mmdetection#7096](https://github.com/open-mmlab/mmdetection/issues/7096)
The key_score may be `None` so it needs to skip the action to save the best checkpoint.

## Modification

Modify the check condition for [eval_hooks.py](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/core/evaluation/eval_hooks.py#L125):
From
```Python
if self.save_best:
    self._save_ckpt(runner, key_score)
```
To
```Python
# the key_score may be `None` so it needs to skip
# the action to save the best checkpoint
if self.save_best and key_score:
    self._save_ckpt(runner, key_score)
```

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by jbwang1997** [comment]
Thank you for your contribution, we will reorganize all README.md in configs in #7051, so the changes of README.md may cause some conflict.

**Review by hhaAndroid — APPROVED** [review]

File: configs/resnest/README.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ It is well known that featuremap attention and multi-path representation are imp
 
 ## Citation
 
+<!-- [ALGORITHM] -->
```
Comment (ZwwWayne) [inline configs/resnest/README.md:19]: The modification is unnecessary

File: configs/regnet/README.md [hunk 2]
```diff
@@ -24,6 +24,8 @@ The pre-trained modles are converted from [model zoo of pycls](https://github.co
 
 ## Citation
 
+<!-- [ALGORITHM] -->
```
Comment (ZwwWayne) [inline configs/regnet/README.md:27]: The modification is unnecessary

**Review by ZwwWayne — APPROVED** [review]

**Comment by OpenMMLab-Assistant003** [comment]
Hi @LuooChen！First of all, we want to express our gratitude for your significant PR in the MMDetection project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤ @LuooChen

**Pull request merged**